.github/workflows/ci.yml

```diff
@@ -160,8 +197,12 @@ jobs:
 
   slack-notification:
     name: Slack Notification
-    if: ${{ failure() && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/release')) }}
+    if: |
+      needs.skip-check.result == 'success' &&
```

Comment:
nit: the `skip-check.result == 'success'` intuitively reads to me "we should skip this job". so the if-statement here initially reads as something like "if the job was successfully skipped and there was a failure and the branches match, then send notification in slack" (which I think is not the intention).

suggestion to reframe the check as when to run a job vs. when to skip a job and maybe rename `skip-check` to something like `check-run-job`, `should-run-job`, `should-run`

I might be misunderstanding this all together, please don't hesitate to let me know

Reply:
Great suggestion, I'll update the name

Reply:
Switched the name to `run-check` in https://github.com/hashicorp/consul-terraform-sync/pull/962/commits/1bf2375d211729674195bdc1355a7dd205fd171c